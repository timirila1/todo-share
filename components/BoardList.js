import React, { Component } from 'react';
import { View, StyleSheet, ScrollView, Vibration } from 'react-native';
import PropTypes from 'prop-types';

import { BoardItem } from './BoardItem';
import { BoardsFooter } from './BoardsFooter';
import { FloatButton } from './FloatButton';
import { routes } from '../services/consts';

export class BoardList extends Component {
    state = { selected: null };

    getBoardItems = () => {
        const { boards } = this.props;
        const { selected } = this.state;

        return boards.map(b => (
            <BoardItem
                board={b}
                onPress={() => this.goToTodos(b)}
                onLongPress={() => this.selectBoard(b)}
                blocked={selected && b.id !== selected.id}
                selected={selected && b.id === selected.id}
                key={b.id}
            />
        ));
    };

    goToTodos = (board) => {
        const { navigation } = this.props;
        const { selected } = this.state;

        if (!selected || board.id === selected.id) {
            this.setState({ selected: null });
            navigation.navigate(routes.Todos, {
                boardId: board.id,
                boardTitle: board.title,
            });
            return;
        }

        this.setState({ selected: null });
    }

    selectBoard = (board) => {
        Vibration.vibrate(100);
        this.setState({ selected: board });
    }

    goToAddBoard = () => {
        this.setState({ selected: null });
        this.props.navigation.navigate(routes.AddBoard);
    }

    afterDelete = () => {
        this.props.updateBoards();
        this.setState({ selected: null });
    }

    render() {
        const { navigation } = this.props;
        const { selected } = this.state;

        return (
            <View style={styles.container}>
                <ScrollView style={styles.container}>
                    <View style={[styles.container, styles.list]}>
                        {this.getBoardItems()}
                    </View>
                </ScrollView>
                {selected && <BoardsFooter
                    selected={selected}
                    navigation={navigation}
                    afterDelete={this.afterDelete}
                />}
                {!selected && <FloatButton
                    name="plus"
                    title="Add Board"
                    onPress={this.goToAddBoard}
                />}
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    list: {
        flexDirection: 'row',
        flexWrap: 'wrap',
        paddingTop: 20,
        paddingLeft: 40,
    },
});

BoardList.propTypes = {
    boards: PropTypes.array,
    navigation: PropTypes.object,
    updateBoards: PropTypes.func,
};
