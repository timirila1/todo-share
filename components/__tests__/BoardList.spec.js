import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { BoardList } from '../BoardList';


it('Render BoardList', async () => {
    const boardsMock = [{
        id: 1,
        title: 'Test title',
        desc: 'Test desc',
    },
    {
        id: 2,
        title: 'Test title',
        desc: 'Test desc',
    }];

    const tree = renderer.create(
        <BoardList
            navigation={{}}
            boards={boardsMock}
        />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
