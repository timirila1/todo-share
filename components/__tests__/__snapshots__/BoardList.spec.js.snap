// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BoardList 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingRight": 20,
          "width": "50%",
        },
        Object {
          "paddingLeft": 40,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 20,
          "minHeight": 250,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#5e1c73",
            "borderRadius": 20,
            "flex": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 24,
              "fontWeight": "300",
              "lineHeight": 36,
            }
          }
        >
          Test title
        </Text>
        <Text
          style={
            Object {
              "color": "rgba(255, 255, 255, 0.8)",
              "fontWeight": "200",
            }
          }
        >
          Test desc
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingRight": 20,
          "width": "50%",
        },
        Object {
          "paddingLeft": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 20,
          "minHeight": 250,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#5e1c73",
            "borderRadius": 20,
            "flex": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 24,
              "fontWeight": "300",
              "lineHeight": 36,
            }
          }
        >
          Test title
        </Text>
        <Text
          style={
            Object {
              "color": "rgba(255, 255, 255, 0.8)",
              "fontWeight": "200",
            }
          }
        >
          Test desc
        </Text>
      </View>
    </View>
  </View>
</View>
`;
